### DISCUSSION TITLE
Migrating from nose to pytest

### DISCUSSION THREAD

**Discussion by mickeypash** [pytest-dev/pytest#8024] [Q&A]
Hello,

I found this page: https://docs.pytest.org/en/reorganize-docs/new-docs/advanced/migratingnose.html?highlight=migrating
I wouldn't mind contributing but I couldn't find it in the Github Docs.

**Comment by The-Compiler** [comment]
Judging from the `reorganize-docs/new-docs`, it looks like this was from an effort to reorganize the entire documentation, which I think was dropped at some point. If it was the thing started at the sprint in 2016, I think @pfctdayelise or @hackebrot might know more (if I'm misremembering, sorry for the noise!).

I don't think there are any plans to continue the reorganizing efforts at the moment - maybe whatever you'd like to contribute would fit in in the [Running tests written for nose](https://docs.pytest.org/en/stable/nose.html) page?

**Comment by mickeypash** [comment]
Thanks!
I just found this tool: https://github.com/pytest-dev/nose2pytest which looks promising!
I also found this wiki that talks about the docs refactor: https://github.com/pytest-dev/pytest/wiki/Docs-refactor

**Reply by The-Compiler** [reply]
Neat, I wasn't even aware of `nose2pytest` being a thing! I'd suggest adding a "Migrating from nose to pytest" section to the [existing docs](https://docs.pytest.org/en/stable/nose.html) which can then point to that tool - would you like to contribute something like that?

**Reply by mickeypash** [reply]
Yes! That would be awesome! I've found some great resources from @magopian and @daneah which might be awesome to include or turn into docs.
- Mathieu's [Switching from nose to py.test at Mozilla](https://agopian.info/presentations/2015_06_djangocon_europe/)
- Dane's [Adopt-a-pytest @PyOhio](https://speakerdeck.com/daneah/adopt-a-pytest)

**Reply by mickeypash** [reply]
I have raised #8048 . Starting with the `nose2pytest` as you recommended with the plan to add the most salient bits from the two talks mentioned above.

**Comment by mickeypash** [comment]
This is a great little table.
I don't know if others like this exist.
All credit goes to @daneah

It would be a nice addition to the docs?

| Convention                  | unittest | pytest |
|-----------------------------|----------|--------|
| Ⓜ test*.py                 | ✅       |        |
| Ⓜ test_*.py                | ✅       | ✅     |
| Ⓜ *_test.py                |          | ✅     |
| Ⓜ *_tests.py               |          |        |
|    Ⓒ  *(unittest.TestCase) | ✅       | ✅    |
|       ⓜ test_*             | ✅       | ✅    |
|    Ⓒ Test*                 |          | ✅      |
|       ⓜ  test_*            |          | ✅      |
|    ⓕ test_*                |          | ✅      |

|Legend||
|---|---------|
| Ⓜ | module |
| Ⓒ | Class |
| ⓜ | method |
| ⓕ | function |

**Reply by daneah** [reply]
Would be important to mention that these can both be told to discover other patterns, so that folks don't think this is what's _possible_ but rather what's _default_.

**Comment by mickeypash** [comment]
I've now added he table with this [change](https://github.com/pytest-dev/pytest/pull/8502)
